tools/vsimporter/xib2nib/NIBWriter.cpp

```diff
@@ -302,21 +302,47 @@ XIBObject *NIBWriter::GetProxyFor(XIBObject *obj)
 }
 
 std::map<std::string, std::string> _g_exportedControllers;
+std::map<std::string, std::string> _g_exportedControllersbyStoryboardId;
+
+void NIBWriter::ExportAllControllers()
+{
+	viewControllerList::iterator cur = XIBObject::_viewControllerNames.begin();
+
+	for (; cur != XIBObject::_viewControllerNames.end(); cur++) {
+		ExportController(*(cur));
+	}
+}
 
 void NIBWriter::ExportController(const char *controllerId)
 {
     std::string controllerName = std::string("UIViewController-") + controllerId;
-    //  Check if we've already written out the controller
-    if (_g_exportedControllers.find(controllerName) != _g_exportedControllers.end()) {
-        return;
-    }
-
-    _g_exportedControllers[controllerName] = controllerName;
+    
 
-    char szFilename[255];
+	char szFilename[255];
     sprintf(szFilename, "UIViewController-%s.nib", controllerId);
 
     XIBObject *controller = XIBObject::findReference(controllerId);
+	UIViewController *uiViewController = (UIViewController *)controller;
+	if (!uiViewController)
+	{
+		//object isn't really a controller
+		return;
+	}
+	
+	//  Check if we've already written out the controller
+	if (_g_exportedControllers.find(controllerId) != _g_exportedControllers.end()) {
+		return;
```

Comment:
If working with valid storyboards, this should never happen. Did not find a logging infrastructure in place in the project so I used a printf statement.